electrum/wallet_db.py

```diff
@@ -1157,6 +1158,23 @@ def _convert_version_60(self):
                 cb['multisig_funding_privkey'] = None
         self.data['seed_version'] = 60
 
+    def _convert_version_61(self):
+        if not self._is_upgrade_method_needed(60, 60):
+            return
+        lightning_payments = self.data.get('lightning_payments', {})
+        for rhash, (amount_msat, direction, is_paid) in list(lightning_payments.items()):
+            new_dataclass_type = {
+                'rhash': rhash,
+                'amount_msat': amount_msat,
+                'direction': direction,
+                'status': is_paid,
+                'min_final_cltv_delta': 147,
+                'expiry_delay': 100 * 365 * 24 * 60 * 60,  # LN_EXPIRY_NEVER
+                'creation_ts': int(time.time()),
+            }
+            lightning_payments[rhash] = new_dataclass_type
+        self.data['seed_version'] = 61
+
```

Comment:
If the wallet db upgrade is only motivated by the use of `dataclass`, then it might not be necessary.
Instead, the `stored_in` decorator can be attached to a constructor, see `UpdateHtlc`.

Reply:
The old db format did not store e.g. `creation_ts` and `expiry_delay`, so I think a db upgrade is needed.

Reply:
The old db format did not store e.g. `creation_ts` and `expiry_delay`, so I think a db upgrade is needed.

Reply:
oh right.